vit-servicing-station-lib/src/db/models/challenges.rs

```diff
@@ -0,0 +1,58 @@
+use crate::db::{schema::challenges, DB};
+use diesel::{ExpressionMethods, Insertable, Queryable};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct Challenge {
+    pub id: i32,
+    pub title: String,
+    pub description: String,
+    pub rewards_total: i64,
+    pub fund_id: i32,
+}
+
+impl Queryable<challenges::SqlType, DB> for Challenge {
+    type Row = (
+        // 0 -> id
+        i32,
+        // 1 -> title
+        String,
+        // 2 -> description
+        String,
+        // 3 -> rewards_total
+        i64,
+        // 4 -> fund_id
+        i32,
+    );
+
+    fn build(row: Self::Row) -> Self {
+        Challenge {
+            id: row.0,
+            title: row.1,
+            description: row.2,
+            rewards_total: row.3,
+            fund_id: row.4,
+        }
+    }
+}
+
+impl Insertable<challenges::table> for Challenge {
+    type Values = (
+        diesel::dsl::Eq<challenges::id, i32>,
+        diesel::dsl::Eq<challenges::title, String>,
+        diesel::dsl::Eq<challenges::description, String>,
+        diesel::dsl::Eq<challenges::rewards_total, i64>,
+        diesel::dsl::Eq<challenges::fund_id, i32>,
+    );
```

Comment:
I think you can just derive this impls in this case.

Reply:
fair point

Reply:
@eugene-babichenko they couldn't finally be derived?
```rust
#[derive(Queryable, Insertable)]
struct Challenge {
    ...
}
```

Maybe you need to add a tag for specifying the table for that.
```rust
#[derive(Insertable, Queryable)]
#[table_name = "challenges"]
```